engines/instance_verification/lib/instance_verification.rb

```diff
@@ -0,0 +1,24 @@
+$LOAD_PATH.push File.expand_path(__dir__, '..')
+
+module InstanceVerification
+  class << self
+    attr_accessor :provider
+  end
+
+  class Exception < RuntimeError; end
+end
+
+module InstanceVerification::Providers
+end
+
+require 'instance_verification/engine'
+require 'instance_verification/provider_base'
+
+providers = Dir.glob(File.join(__dir__, 'instance_verification/providers/*.rb'))
+
+raise 'Too many instance verification providers found' if providers.size > 1
```

Comment:
This forces anyone to implement verification in one file or spread the verification into multiple directories. I am OK with this, we just have to be aware of the restriction and communicate when we re-write the setup documentation.